strats.md

```diff
@@ -45,3 +47,411 @@ Execution of a strat may have an effect of clearing one or more [obstacles](regi
+- _minTiles_: The minimum amount of effective runway tiles in other room needed for this strat.
+
+A `comeInStutterCharging` condition must match with a `leaveWithRunway` condition on the other side of the door, which must have an "air" environment. A match comes with the following implicit requirements for actions to be performed in the previous room:
+- The tech `canStutterWaterShineCharge`, which includes a requirement for the `SpeedBooster` item.
+- If the previous room is heated, then heat frame requirements are included based on `minTiles`, in the same way as for a `comeInRunning` requirement.
+
+#### Example
+```json
+{
+  "name": "Come In Stutter Charging",
+  "notable": false,
+  "entranceCondition": {
+    "comeInStutterCharging": {}
+  },
+  "requires": [
+    {"shinespark": {"frames": 60}}
+  ]
+}
+```
+
+### Come In With Bomb Boost
+
+A `comeInWithBombBoost` entrance condition indicates that Samus must come into the room with a horizontal bomb boost. This object has no properties.
+
+A `comeInWithBombBoost` condition must match with a `leaveWithRunway` condition on the other side of the door, which must have an "air" environment. A match comes with the following implicit requirements for actions to be performed in the previous room:
+- A requirement `h_canBombThings`, to be able to use Bombs or a Power Bomb.
+- If the previous room is heated, a requirement of `{"heatFrames": 100}` is included, for positioning, placing the bomb, and waiting for it to detonate.
+
+#### Example
+```json
+{
+  "name": "Come In With Bomb Boost",
+  "notable": false,
+  "entranceCondition": {
+    "comeInWithBombBoost": {}
+  },
+  "requires": []
+}
+```
+
+### Come In With Door Stuck Setup
+
+A `comeInWithDoorStuckSetup` entrance condition indicates that Samus must enter the room in a way that allows getting stuck in the door as it closes. This object has no properties.
+
+A `comeInWithDoorStuckSetup` condition must match with a `leaveWithRunway` condition on the other side of the door. A match comes with the following implicit requirements for actions to be performed in the previous room:
+- If the previous room is heated, a requirement of `{"heatFrames": 100}` is included, for positioning and executing the door stuck setup. Note that if X-Ray Scope is available (and it always should be for a strat with this condition, since the only known application of door stuck setups is for X-Ray climbing), then minimizing heat damage is made easier by using X-Ray to buffer the positioning near the door and to turn around in place.
+- If the previous room is to the left, then a tech requirement `canStationarySpinJump` is included.
+- If the previous room is to the right, then a tech requirement `canRightSideDoorStuck` is included.
+- If the previous door environment is water and is to the right, then a requirement `{"or": ["Gravity", "canRightSideDoorStuckFromWater"]}` is included.
```

Comment:
`canShinesparkDeepstuck` doesn't have that many combinations of rooms that allow it to work, without spikesuit/bluesuit.
`GMode` deepstuck would be setup in room without this entrance condition.

Reply:
I assume we'll probably use a `comeInCharged` entrance condition for shinespark deep stuck strats. We could make something more specific but I'm not sure if it's needed?

Reply:
`comeInCharged` would be a good way to handle it.